crates/typed-store/src/rocks/mod.rs

```diff
@@ -1734,6 +1739,15 @@ where
         Ok(())
     }
 
+    /// Same as `clear` but allows caller to guard the critical section with a
+    /// a read write lock. It is caller's responsibility to use the lock around
+    /// other db operations to prevent the race condition from happening
+    #[instrument(level = "trace", skip_all, err)]
+    fn safe_clear(&self, rw_lock: Arc<RwLock<()>>) -> Result<(), TypedStoreError> {
```

Comment:
Couldn't the caller just take the lock themselves before calling this? Also, since the bug we found can affect any use of the cf in question from anywhere, this doesn't seem like a very practical way to make things safe.

I think a better approach is to just rename clear to unsafe_clear, document the issue, and add a `delete_all` method that uses a range delete.

Reply:
Yeah I agree the caller can themselves do the locking around db operations and there is no extra value we are providing here. I wasn't sure what @lxfind had in mind for safe_clear but delete_all is also ok (doesn't have the same semantics but is probably the next best thing to do)

Reply:
@mystenmark can you take another look, thanks.